### ISSUE TITLE
Best practice for using RM to control stuff *not rendering directly by React*?

### ISSUE THREAD

**Issue by zth** [chenglou/react-motion#213]
Hi!

I recently encountered a situation where I wanted to use React Motion to animate the scroll inside an element. The element is basically a large map with an element wrapped around it that exposes only a small part of the map, and lets you scroll/pan around to explore it. So, what I wanted to do was using React Motion to animate panning/scrolling back and forth between various set points. Here's how I ended up solving it (alot of code omitted):

    render() {

		...

		const panElement = this.panElement; // This is the DOM-node for the pan element which I'm scrolling

		return (
			<div>
				<Motion 
					defaultStyle={{
						x: this.state.x1, 
						y: this.state.y1
					}} 
					style={{
						x: shouldMove ? spring( this.state.x2 ) : this.state.x2, 
						y: shouldMove ? spring( this.state.y2 ) : this.state.y2
					}}>
					{ value => {

						if( shouldMove ) {
							panElement.scrollLeft = value.x;
							panElement.scrollTop = value.y;
						}

						return (
							<div />
						);
					}}
				</Motion>
				<ElementPan ref="panElement">
                    ....
				</ElementPan>
			</div>
		);

So, basically, my question is: how do I do this better/smarter with React Motion? This _works_, but I'm sure I could do this in a smarter way, because it feels like I'm just hacking McGyver style here with returning an empty div.
Ideally, I'd like to be able to use this outside of the render function all together, and just feed a function a start/end value for both x and y, and have it animate that scroll without even touching the components state.

I should also note that this is literally one of the first things I've written with React Motion (I love it btw), so I appologize if I'm missing something obvious.

Thanks in advance for any help, and thanks for React Motion in general, I'm really excited about this!

**Comment by chenglou** [comment]
You should probably wrap the panElement logic into a declarative wrapper:
```jsx
<ElementPan ref="panElement" scrollLeft={...}>...
```
In `componentDidUpdate` of ElementPan, there you set the imperative `scrollLeft` to `this.props.scrollLeft`. From the outside it still looks declarative.

After that, you can move ElementPan into Motion.

Does that work?

**Comment by nkbt** [comment]
You may also try adding debounce to onScroll (like 50ms), keep values in state, and then use Motion to move things smoothly.

**Comment by chenglou** [comment]
This should work. Closing! If you have further questions, we're all on [Discord](https://discordapp.com/invite/0ZcbPKXt5bYzmcI0) so ask there =). Thanks!
